packages/client/src/debounce.test.tsx

```diff
@@ -0,0 +1,23 @@
+import debounce from "./debounce"
+
+jest.useFakeTimers()
+
+describe("debounce", () => {
+  let func: jest.Mock
+  let debouncedFunc: Function
+
+  beforeEach(() => {
+    func = jest.fn()
+    debouncedFunc = debounce(func, 3000, false)
+  })
+
+  it("executes after 3 sec delay when called multiple times", () => {
+    for (let i = 0; i < 300; i++) {
+      debouncedFunc()
+    }
+
+    jest.runAllTimers()
+
+    expect(func).toBeCalledTimes(1)
+  })
+})
```

Comment:
I would like a test for this scenario: every time `debouncedFunc` is called it is called with a different argument. Verify that in any conditions `func` is eventually called with the same argument as the _last_ call to `debouncedFunc`.

I would also like to test that if `debouncedFunc` is called many times over a period longer than 3 seconds that `func` is called every 3 seconds.